lib/common/alarm/ScaleAlarms.ts

```diff
@@ -0,0 +1,143 @@
+
+  /**
+   * A multiplication factor to apply to the threshold of the cloned alarms.
+   * @default - 1.0
+   */
+  readonly thresholdMultiplier?: number;
+
+  /**
+   * A multiplication factor to apply to the datapointsToAlarm property of the cloned alarms.
+   * @default - 1.0
+   */
+  readonly datapointsToAlarmMultiplier?: number;
+
+  /**
+   * A multiplication factor to apply to the `evaluationPeriods` property of the cloned alarms.
+   * @default - Same as datapointsToAlarmMultiplier.
+   */
+  readonly evaluationPeriodsMultiplier?: number;
+}
+
+/**
+ * Creates a function that clones alarms to be more scaled using the multiplication factors
+ * provided. This function can be provided to the `cloneAlarms` method of
+ * {@link MonitoringFacade}.
+ *
+ * This can be used with scaling factors less than 1.0 to create more aggressive alarms, which
+ * can be useful to provide early detection and trigger change rollbacks.
+ *
+ * @param props The properties for configuring the function.
+ * @returns A function that clones alarms and applies scaling factors.
+ */
+export function ScaleAlarms(props: ScaleAlarmsProps): AlarmCloneFunction {
+  return function (alarm: AlarmWithAnnotation) {
+    let thresholdMultiplierToUse: number = 1;
+    if (props.thresholdMultiplier) {
+      const comparisonOperator =
+        alarm.alarmDefinition.addAlarmProps.comparisonOperator;
+      switch (comparisonOperator) {
+        case ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD:
+        case ComparisonOperator.GREATER_THAN_THRESHOLD:
+          thresholdMultiplierToUse = props.thresholdMultiplier;
+          break;
+        case ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD:
+        case ComparisonOperator.LESS_THAN_THRESHOLD:
+          thresholdMultiplierToUse = 1.0 / props.thresholdMultiplier;
+          break;
+        case ComparisonOperator.GREATER_THAN_UPPER_THRESHOLD:
+        case ComparisonOperator.LESS_THAN_LOWER_THRESHOLD:
+        case ComparisonOperator.LESS_THAN_LOWER_OR_GREATER_THAN_UPPER_THRESHOLD:
+          // Don't modify thresholds for anomoly detection models.
+          thresholdMultiplierToUse = 1;
+          break;
+        default:
+          throw new Error(
+            `Comparison operator ${comparisonOperator} is not supported by ScaleAlarms.`,
+          );
+      }
+    }
+    let cloneThreshold =
+      alarm.alarmDefinition.addAlarmProps.threshold * thresholdMultiplierToUse;
+
+    let cloneEvaluationPeriods =
+      alarm.alarmDefinition.evaluationPeriods *
+      (props.evaluationPeriodsMultiplier ??
+        props.datapointsToAlarmMultiplier ??
+        1);
+    let cloneDatapointsToAlarm =
+      alarm.alarmDefinition.datapointsToAlarm *
+      (props.datapointsToAlarmMultiplier ?? 1);
+
+    // If the tightened datapoints to alarm is now so small that we can't effectively alarm on it,
+    // reduce the metric period to the next smallest duration and scale up the datapoint count accordingly.
```

Comment:
The docs for `ScaleAlarms` should be more transparent about this sort of clamping behaviour, otherwise it may be surprising.

Reply:
More specifically -- can we include this in the main JSDocs for `ScaleAlarms` rather than just in a comment here?